provider/aci.go

```diff
@@ -1639,38 +1640,39 @@ func getProbe(probe *v1.Probe, ports []v1.ContainerPort) (*aci.ContainerProbe, e
 }
 
 func (p *ACIProvider) getAzureFileCSI(volume v1.Volume, namespace string) (*aci.Volume, error) {
-	azureSource := &aci.AzureFileVolume{
-		ReadOnly: *volume.CSI.ReadOnly,
-	}
-	if volume.CSI.NodePublishSecretRef != nil && volume.CSI.NodePublishSecretRef.Name != "" {
-
-		// Set the storage account name and key
-		secret, err := p.resourceManager.GetSecret(volume.CSI.NodePublishSecretRef.Name, namespace)
-		if err != nil {
-			return nil, errors.Wrap(err, fmt.Sprintf("The secret %s for AzureFile CSI driver %s is not found", volume.CSI.NodePublishSecretRef.Name, volume.Name))
-		}
+	azureSource := &aci.AzureFileVolume{}
 
-		if secret == nil {
-			return nil, fmt.Errorf("Getting secret for AzureFile CSI driver %s returned an empty secret", volume.Name)
-		}
+	if volume.CSI.VolumeAttributes != nil && len(volume.CSI.VolumeAttributes) != 0 {
+		secretName := volume.CSI.VolumeAttributes[azureFileSecretName]
+		if secretName != "" {
+			secret, err := p.resourceManager.GetSecret(secretName, namespace)
 
-		azureSource.StorageAccountName = string(secret.Data["azurestorageaccountname"])
-		azureSource.StorageAccountKey = string(secret.Data["azurestorageaccountkey"])
+			if err != nil {
+				return nil, errors.Wrap(err, fmt.Sprintf("the secret %s for AzureFile CSI driver %s is not found", secretName, volume.Name))
+			}
 
-		// Set shareName
-		if volume.CSI.VolumeAttributes != nil {
-			if shareName, ok := volume.CSI.VolumeAttributes[azureFileShareName]; ok {
-				azureSource.ShareName = shareName
+			if secret == nil {
+				return nil, fmt.Errorf("getting secret for AzureFile CSI driver %s returned an empty secret", volume.Name)
 			}
+			// Set the storage account name and key
+
+			azureSource.StorageAccountName = string(secret.Data["azurestorageaccountname"])
+			azureSource.StorageAccountKey = string(secret.Data["azurestorageaccountkey"])
```

Comment:
it's better define `azurestorageaccountkey` and `azurestorageaccountname` as const vars since they are referenced multiple times.